src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/DumpTests.cs

```diff
@@ -74,5 +79,75 @@ public Task DumpTest(DiagnosticPortConnectionMode mode, DumpType type)
                     }
                 });
         }
+
+        [ConditionalTheory(typeof(TestConditions), nameof(TestConditions.IsDumpSupported))]
+        [InlineData(DiagnosticPortConnectionMode.Connect, DumpType.Mini)]
+        public Task DiagSessionTest(DiagnosticPortConnectionMode mode, DumpType type)
+        {
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                mode,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (runner, client) =>
+                {
+                    int processId = await runner.ProcessIdTask;
+
+                    PackageMode mode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? PackageMode.DiagSession : PackageMode.IncludeDacDbi;
```

Comment:
We have a variable named `mode` in the encompassing context that has a different meaning (`DiagnosticPortConnectionMode`); one of these should probably be changed.